core/src/main/java/org/apache/iceberg/actions/FileRewriter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.iceberg.ContentFile;
+import org.apache.iceberg.ContentScanTask;
+
+/**
+ * A class for rewriting content files.
+ *
+ * @param <T> the Java type of tasks to read content files
+ * @param <F> the Java type of content files
+ */
+public interface FileRewriter<T extends ContentScanTask<F>, F extends ContentFile<F>> {
```

Comment:
I created a separate hierarchy because I did not follow the exact API.
  - I replaced `RewriteStrategy options(map)` with `void init(map)`. In order to support proper method chaining with inheritance, we need to parametrize the strategy and rewriter with `ThisT`, like we do in `Scan`, for instance. That was not done and method chaining was partially broken in existing strategies. To simplify the API, I went for the approach we use in catalogs and `FileIO` and added `void init(map)` instead.
  - I did not make the new interface serializable as our strategies are not used in a distributed fashion. They can’t be serialized in their current form too (they have non-serializable fields). If we want to support that functionality in the future, we may add it later. We should either mark rewriters serializable and support that or don’t do that.